src/reactor/block_reporter.rs

```diff
@@ -0,0 +1,127 @@
+    /// * `event_loop` - The reactor's event loop to handle the tasks spawned by this observer.
+    /// * `block_chain` - Block chain object, where block reporting will happen.
+    pub fn new(
+        block_store: Arc<Contract<Http>>,
+        from: Address,
+        event_loop: Box<tokio_core::reactor::Handle>,
+        block_chain: Arc<Ethereum>,
+    ) -> Self {
+        BlockReporter {
+            block_store,
+            from,
+            event_loop,
+            block_chain,
+        }
+    }
+}
+
+impl React for BlockReporter {
+    /// Defines how different reactor will react on block observation.
+    ///
+    /// # Arguments
+    ///
+    /// * `block` - The observed block.
+    fn react(&self, block: &Block) {
+        info!("Reporting block for number {:?} ", block.number);
+
+        let encoded_block = rlp::encode(block);
+        let block_hash = block.hash();
+        let block_store = Arc::clone(&self.block_store);
+        let block_chain = &self.block_chain;
+        let from = self.from;
+
+        // Unlocking account with zero time interval, which unlocks account till node is up.
+        let call_future = block_chain.unlock_account(Some(0)).then(move |_| {
+            block_store
+                .query(
+                    "isBlockReported",
+                    block_hash,
+                    from,
+                    Options::default(),
+                    None,
+                ).then({
+                    move |result: Result<bool, web3::contract::Error>| match result {
+                        Ok(is_reported) => if is_reported {
+                            Either::A(Ok(()).into_future())
+                        } else {
+                            Either::B(
+                                block_store
+                                    .call(
+                                        "reportBlock",
+                                        encoded_block,
+                                        from,
+                                        Options::with(|opt| {
+                                            opt.gas = Some(REPORT_BLOCK_ESTIMATED_GAS.into())
+                                        }),
+                                    ).then(move |tx| {
+                                        info!("Block reported got tx: {:?}", tx);
+                                        Ok(())
+                                    }).map_err(|error: Error| {
+                                        error!("Error reporting block {:?}", error)
+                                    }),
+                            )
+                        },
+                        Err(error) => {
+                            error!(
+                                "Error while checking if block is already reported{:?}",
+                                error
+                            );
+                            // Event loop spawn expects certain types. It doesn't support err types.
```

Comment:
Is it documented somewhere that reactors should do the error handling internally? (fire and forget)